### PR TITLE
[core][python] Add optional Bloom filters to BTree indexes

### PR DESCRIPTION
**Pull request by JingsongLi**
### Purpose

Add optional Bloom filter support to BTree global indexes while keeping it disabled by default.

- Add `btree-index.bloom-filter.enabled` with default `false`.
- Write and read Bloom filters for Java and Python BTree index files.
- Use exact point lookup for equality and `IN` predicates so Bloom-negative keys avoid data-block reads.
- Keep the Java and Python hash and bit layouts compatible.
- Document the new option.

### Tests

- `mvn -pl paimon-common -am -DwildcardSuites=none -DfailIfNoTests=false -Dtest=BTreeBloomFilterTest,BTreeIndexOptionsTest test`
- Python BTree suite: 36 passed, 4 skipped
- Python global-index suite: 68 passed, 19 subtests passed
- Flake8 on changed Python files

### CODE REVIEW

File: paimon-python/pypaimon/globalindex/btree/btree_index_writer.py [hunk 1]
```diff
@@ -100,15 +106,23 @@ def finish(self) -> bytes:
...
         self._block_size = block_size
         self._data_block_writer = _BlockWriter()
         self._index_block_writer = _BlockWriter()
         self._last_key: Optional[bytes] = None
+        self._bloom_hashes = array('i') if bloom_filter_enabled else None
 
     def put(self, key: bytes, value: bytes) -> None:
         self._data_block_writer.add(key, value)
+        if self._bloom_hashes is not None:
+            self._bloom_hashes.append(murmur_hash_bytes(key))
```
Comment (leaves12138) [inline paimon-python/pypaimon/globalindex/btree/btree_index_writer.py:125]: Update after discussion: NaN handling is outside the agreed scope of this PR. The concern previously raised here is withdrawn as a blocker; no change is requested for it in this PR.

**Review by leaves12138 — COMMENTED** [review]
Review update: the previously reported NaN edge case is outside the agreed scope and is not a blocker. The focused Java suite passed (340 tests), and the Python BTree/global-index suite passed (109 tests, 4 skipped). Additional Java/Python interoperability checks found no other blocking issues within the agreed scope.

**Review by leaves12138 — APPROVED** [review]
LGTM within the agreed review scope. Verified the default-disabled behavior, equality/IN lookup paths, duplicate-key row IDs, null-only indexes, and Java/Python interoperability. The focused Java suite passed (340 tests), and the Python BTree/global-index suite passed (109 tests, 4 skipped). The previously discussed out-of-scope edge case is not a blocker for this PR.
